plugins/modules/aci_route_control_match_dest.py

```diff
@@ -0,0 +1,312 @@
+                "descr": "Production environment"
+            }
+        }
+    }
+previous:
+  description: The original configuration from the APIC before the module has started
+  returned: info
+  type: list
+  sample:
+    [
+        {
+            "fvTenant": {
+                "attributes": {
+                    "descr": "Production",
+                    "dn": "uni/tn-production",
+                    "name": "production",
+                    "nameAlias": "",
+                    "ownerKey": "",
+                    "ownerTag": ""
+                }
+            }
+        }
+    ]
+proposed:
+  description: The assembled configuration from the user-provided parameters
+  returned: info
+  type: dict
+  sample:
+    {
+        "fvTenant": {
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        subject_name=dict(type="str"),
+        ip=dict(type="str"),
+        aggregate=dict(type="bool"),
+        greater_than_mask=dict(type="int", aliases=["greater_than", "gt"]),
+        less_than_mask=dict(type="int", aliases=["less_than", "lt"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "subject_name", "ip"]],
+            ["state", "present", ["tenant", "subject_name", "ip"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    tenant = module.params.get("tenant")
+    subject_name = module.params.get("subject_name")
+    ip = module.params.get("ip")
+    aggregate = aci.boolean(module.params.get("aggregate"))
+    greater_than = module.params.get("greater_than_mask")
+    less_than = module.params.get("less_than_mask")
+    state = module.params.get("state")
+
+    if ip is not None:
+        mask = int(ip.split("/")[1])
```

Comment:
I've added a basic regex test for this, to try and ensure valid IP data (including ensuring that the prefix length is present). The IPv6 regex is a little basic, since a full regex for IPv6 that matches all valid IPv6 and rejects all invalid ones winds up being enormous.